test/ecto/query_test.exs

```diff
@@ -1133,4 +1133,22 @@ defmodule Ecto.QueryTest do
       assert inspect(reverse_order(q)) == inspect(order_by(q, desc: :id))
     end
   end
+
+  describe "sigil_f/1" do
+    test "builds a fragment without interpolation" do
+      assert inspect(select("t", ~f"NOW()")) == inspect(select("t", fragment("NOW()")))
+    end
+
+    test "builds a fragment with one interpolation" do
+      assert inspect(select("t", [t], ~f"LOWER(#{t.email})")) ==
+               inspect(select("t", [t], fragment("LOWER(?)", t.email)))
+    end
+
+    test "builds a fragment with multiple interpolation" do
+      date = ~D[2020-01-01]
+
+      assert inspect(select("t", [t], ~f"GREATEST(#{t.date}, #{^date})")) ==
+               inspect(select("t", [t], fragment("GREATEST(?, ?)", t.date, ^date)))
+    end
+  end
```

Comment:
We should probably add a test for interpolating the same variable multiple times. It also prompts the discussion: does doing this create multiple positional arguments, repeatedly interpolating the same value? If so, are we ok with that, what are the trade-offs? Are there better ways? etc

Reply:
I don't think we need a test for interpolating the same variable multiple times as there is no special case for that. We just translate everything to `?` and add the interpolation as argument to `fragment`.

Of course this means we also inherit all the same "semantics" from `fragment`,  which include this repetition and the only current way of fixing that would be using either subqueries or `dynamic`.

IMHO, even though I agree that these are some pain points, I don't know if it is directly related to this PR.

Reply:
The idea from the mailing list tries to address that, the example given by Phillip was this:

```elixir
fragment(
  ":service_year < :year OR (:service_year = :year AND :service_month <= :month)",
  service_year: service.invoice_year,
  service_month: service.invoice_month,
  year: ^max_year,
  month: ^max_month
)
```

The challenge however was that this would require a much more complex parser for the query string.

Maybe we could do something like this

```elixir
fragment(
  "#{arg(:service_year)} < #{^max_year} OR (#{arg(:service_year)} = #{^max_year} AND #{service.invoice_month} <= #{^max_month})",
  service_year: service.invoice_year
)
```

(If we followed the idea of just adding interpolation to `fragment` instead of adding a sigil).

However, this would require adding either a reserved function name like `arg` or  reserve another unary operator, which is probably harder.

This example is not great, of course, but even if `service_year` was another complex fragment, I'm not sure what is the benefit of that vs just doing a subquery to compute and then re-use other than giving more options.

But yeah, the amount of things I wrote show that this is a deeper problem then just allowing interpolation so probably should be discussed separately.